crates/core/prelude/text-encoding.js

```diff
@@ -4,7 +4,11 @@
 
     class TextDecoder {
         constructor(label = "utf-8", options = {}) {
-            if (label !== "utf-8") {
+            label = label.trim().toLowerCase();
+            if (
+                label !== "utf-8" && label !== "utf8" && label !== "unicode-1-1-utf-8"
+                && label !== "unicode11utf8" && label !== "unicode20utf8" && label !== "x-unicode20utf8"
+            ) {
```

Comment:
Just to make it slightly more readable, how about this?

```suggestion
            const acceptedLabels = ["utf-8", "utf8", "unicode-1-1-utf-8", "unicode11utf8", "unicode20utf8", "x-unicode20utf8"];
            if (!acceptedLabels.includes(label)) {
```